### PR TITLE
Add partial for admin items#new and items#edit, all tests passing, paperclip still not implemented

### PR DESCRIPTION
**Pull request by cmacaulay**
👍 
- All tests pass for admin updating and creating an item 
- Just need to implement paperclip 
- Also adds 12factor gem for production, something that was part of mix master

Notes/Questions:
- Once we have an admin_item_path should we link to it on the admin_items page? If yes, we could make the tests for these features even more robust

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -1,13 +1,20 @@
 class Item < ApplicationRecord
+  # attr_reader :image_remote_url
   validates :title, :description, :price, presence: true
   validates :title, :description, uniqueness: true
   belongs_to :category
   has_many :order_items
   has_many :orders, through: :order_items
-
+  has_attached_file :image # styles: {medium: "200x200", thumb: "100x100"}
```
Comment (meyerhoferc) [inline app/models/item.rb]: 👍

File: Gemfile [hunk 2]
```diff
@@ -20,7 +20,8 @@ gem 'uglifier', '>= 1.3.0'
 gem 'coffee-rails', '~> 4.2'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
-
+gem 'rails_12factor', group: :production
+gem 'paperclip'
```
Comment (meyerhoferc) [inline Gemfile]: 👍

File: app/views/admin/items/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  <div class="field drop-down">
+    <%= f.label :category %>
+    <select class="browser-default" name= "item[category_id]">
+      <% @categories.each do |category| %>
+          <option value=<%= category.id %>><%= category.name %></option>
+      <% end %>
+    </select></br>
+  </div>
+    <%= f.submit "Submit", class: "waves-effect waves-light btn" %>
```
Comment (meyerhoferc) [inline app/views/admin/items/_form.html.erb:30]: 👍

File: db/migrate/20170226223536_add_attachment_image_to_items.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class AddAttachmentImageToItems < ActiveRecord::Migration
+  def self.up
+    change_table :items do |t|
+      t.attachment :image
+    end
+  end
+
+  def self.down
+    remove_attachment :items, :image
```
Comment (meyerhoferc) [inline db/migrate/20170226223536_add_attachment_image_to_items.rb:9]: 👍

File: spec/features/admin/items/admin_can_add_item_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+     click_on "Create New Item"
+
+     fill_in "item[title]", with: "Test item"
+     fill_in "item[description]", with: "Test description"
+     fill_in "item[price]", with: "40"
+     select category.name, from: "item[category_id]"
+     click_on "Submit"
+
+     expect(current_path).to eq(admin_items_path)
+     expect(page).to have_content("Test item")
```
Comment (meyerhoferc) [inline spec/features/admin/items/admin_can_add_item_spec.rb:20]: 🎉

File: spec/features/admin/items/admin_can_edit_item_spec.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    fill_in "item[description]", with: "Test description"
+    fill_in "item[price]", with: "40"
+    select category.name, from: "item[category_id]"
+    click_on "Submit"
+
+    expect(current_path).to eq(admin_items_path)
+    expect(page).to have_content("Test item")
+    expect(page).to_not have_content(item.title)
+    expect(page).to have_content("Test description")
+    expect(page).to_not have_content(item.description)
```
Comment (meyerhoferc) [inline spec/features/admin/items/admin_can_edit_item_spec.rb:26]: 🔥 for testing

**Review by meyerhoferc — APPROVED** [review]
Love the testing here, looking forward to getting through paperclip docs eventually. In response to your question, I just finished tests for an item show page for an admin and that PR will be up after Charlotte pushes up her latest changes. There's some formatting she did that I want to pull down so that things are uniform.

File: app/views/admin/items/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+<h4>Create New Item</h4>
+<%= render partial: 'form' %>
```
Comment (CjMoore) [inline app/views/admin/items/new.html.erb:2]: Partials 👍

**Review by CjMoore — APPROVED** [review]
Looks great. Merging now.

**Pull request merged**